tasks/create.yml

```diff
@@ -1,31 +1,36 @@
 ---
-- name: Ensure the BeeGFS package repo is defined
-  get_url:
-    url: "{{ beegfs_repo_url }}"
-    dest: "{{ beegfs_repo_file }}"
-  become: true
-
-- name: Load package repo key
+- name: Add BeeGFS repos is beegfs_add_repos is true
   block:
-    - name: add apt key for debian family machines
-      apt_key:
-        state: present
-        url: "{{ beegfs_repo_key }}"
-      when: ansible_os_family == "Debian"
-    - name: add rpm key for redhat family machines
-      rpm_key:
-        state: present
-        key: "{{ beegfs_repo_key }}"
-      when: ansible_os_family == "RedHat"
-  become: true
 
-- name: Refresh apt repos
-  apt:
-    name: base-files
-    state: present
-    update_cache: true
-  become: true
-  when: ansible_os_family == "Debian"
+  - name: Ensure the BeeGFS package repo is defined
+    get_url:
+      url: "{{ beegfs_repo_url }}"
+      dest: "{{ beegfs_repo_file }}"
+    become: true
+
+  - name: Load package repo key
+    block:
+      - name: add apt key for debian family machines
+        apt_key:
+          state: present
+          url: "{{ beegfs_repo_key }}"
+        when: ansible_os_family == "Debian"
+      - name: add rpm key for redhat family machines
+        rpm_key:
+          state: present
+          key: "{{ beegfs_repo_key }}"
+        when: ansible_os_family == "RedHat"
+    become: true
+
+  - name: Refresh apt repos
+    apt:
+      name: base-files
+      state: present
+      update_cache: true
+    become: true
+    when: ansible_os_family == "Debian"
+
+  when: beegfs_add_repos
```

Comment:
just a note: adding the `| bool` filter allows you to set this variable to `yes`, `no`, or other boolean strings.

Reply:
I always use `true` or `false` because the default yamlint rules in molecule include the "thuthy" rule:
https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.truthy

in any case I will add the filter